src/html/html-document.ts

```diff
@@ -101,6 +101,22 @@ export class ParsedHtmlDocument extends ParsedDocument<ASTNode, HtmlVisitor> {
       }
     };
   }
+
+  stringify() {
+    // TODO(issues/315): serialize inline documents correctly.
+    let result = parse5.serialize(this.ast);
+
+    // Strip out inferred boilerplate nodes that are injected.
```

Comment:
hmm... they're not always injected. What about checking the original source to see if they are there?